javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -67,19 +101,325 @@ class Properties(protected val defaults: Properties)
   }
 
   def list(out: PrintStream): Unit = {
-    out.println("-- listing properties --")
-    entrySet().asScala.foreach { entry => out.println(format(entry)) }
+    list(new PrintWriter(new OutputStreamWriter(out), true))
   }
 
   def list(out: PrintWriter): Unit = {
     out.println("-- listing properties --")
     entrySet().asScala.foreach { entry => out.println(format(entry)) }
   }
 
+  def store(out: OutputStream, comments: String): Unit = {
+    if (out == null) {
+      throw new NullPointerException()
+    }
+    // https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.InputStream-
+    val writer = new OutputStreamWriter(out, "ISO8859_1")
+    store(writer, comments)
+  }
+
+  def store(writer: Writer, comments: String): Unit = {
+    if (writer == null) {
+      throw new NullPointerException()
+    }
+
+    if (comments != null) {
+      writeComments(writer, comments)
+    }
+
+    writer.write('#')
+    writer.write(new Date().toString)
+    writer.write(System.lineSeparator)
+
+    val buffer = new StringBuilder(200)
+    entrySet().asScala.foreach { entry =>
+      val key = entry.getKey.asInstanceOf[String]
+      dumpString(buffer, key, true, false)
+      buffer.append('=')
+      dumpString(buffer, entry.getValue.asInstanceOf[String], false, false)
+      buffer.append(System.lineSeparator)
+      writer.write(buffer.toString)
+      buffer.setLength(0)
+    }
+    writer.flush()
+  }
+
+  @deprecated("", "") def save(out: OutputStream, comments: String): Unit =
+    store(out, comments)
+
+  private val NONE     = 0
+  private val SLASH    = 1
+  private val UNICODE  = 2
+  private val CONTINUE = 3 // when \r is encountered looks for next \n
+  private val KEY_DONE = 4
+  private val IGNORE   = 5
+  private lazy val nextCharMap =
+    SMap('b' -> '\b', 'f' -> '\f', 'n' -> '\n', 'r' -> '\r', 't' -> '\t')
+
+  private def isEbcdic(in: BufferedInputStream): Boolean = {
+    var b: Byte = 0
+    while ({ b = in.read.toByte; b != -1 }) {
+      if (b == 0x23 || b == 0x0a || b == 0x3d) { // ascii: newline/#/=
+        return false
+      }
+      if (b == 0x15) { // EBCDIC newline
+        return true
+      }
+    }
+    //we found no ascii newline, '#', neither '=', relative safe to consider it
+    //as non-ascii, the only exception will be a single line with only key(no value and '=')
+    //in this case, it should be no harm to read it in default charset
+    false
+  }
+
+  private def loadImpl(reader: Reader): Unit = {
+    var mode           = NONE
+    var unicode        = 0
+    var count          = 0
+    var nextChar: Char = 0
+    var buf            = new Array[Char](80)
+    var offset         = 0
+    var keyLength      = -1
+    val br             = new BufferedReader(reader)
+
+    @tailrec def processNext(isFirstChar: Boolean): Unit = {
+      val intVal = br.read()
```

Comment:
Consider using `br.readLine()`, then work on that string. `br.readLine()` already takes care of handling `\r`, `\n` and `\r\n`, so you don't have to mess with `isFirstChar` and `CONTINUE`. You would only have to parse line by line.

Reply:
I went with a line oriented approach - multi-line for continued values.